### ISSUE TITLE
error: stack overflow (possible infinite recursion)

### ISSUE THREAD

**Issue by seghers** [pta2002/nixvim#135]
After upgrading the nixvim flake I get the following error during `nixos-rebuild build` 
```
"locked": {
        "lastModified": 1674327176,
        "narHash": "sha256-q+HpspHrO5XZnMShBJdLHjHVsCoaWN/Tis7OzRwaF3w=",
        "lastModified": 1674160777,
        "narHash": "sha256-ozE4FVz+UB+5pTiNjhYanS/y52lwXhv8mWzAugbIHak=",
        "owner": "pta2002",
        "repo": "nixvim",
        "rev": "23327700e2a16a3434920d26cb61d90b48478fe7",
        "rev": "d2f714c40ec9bcb20f91a8124f33c81f0576bb68",
        "type": "github"
      },
```
The second one works, the first gives the error.

`error: stack overflow (possible infinite recursion)`

**Comment by pta2002** [comment]
Can you run the rebuild with `--show-trace` and paste the output? Thanks

**Comment by seghers** [comment]
nixos-rebuild build --flake .#mydevice --show-trace

```
building the system configuration...
evaluating derivation 'git+file:///pathToConfig#nixosConfigurations."mydevice".config.system.build.toplevel'error: stack overflow (possible infinite recursion)t
```

**Comment by Alexnortung** [comment]
What is in your config?

**Comment by seghers** [comment]
```
programs.nixvim.plugins.lsp.servers = {
        # rust-analyzer.enable = true;
}
```
commenting out rust-analyzer seems to fix it.

**Comment by Alexnortung** [comment]
Alright, I am not really sure what is going on, maybe @traxys has an idea :)

**Comment by traxys** [comment]
Found the problem, published a fix in #137. It seems that my lazyness came and bit me, I needed to exclude some options from the LSP settings.
